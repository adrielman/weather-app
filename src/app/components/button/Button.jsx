import React from "react";
import styles from "./Button.module.scss";
import classNames from "classnames";

export const ButtonType = {
  Danger: "danger",
  Default: "default",
  Primary: "primary"
};

export const Button = ({
  onClick,
  children,
  compact = false,
  nospacing = false,
  type = ButtonType.Default,
  style
}) => {
  const clickHandler = event => {
    event.preventDefault();
    onClick();
  };
  let buttonStyle = classNames(styles.button, {
    [styles.compact]: compact,
    [styles.nospacing]: nospacing,
    [styles.danger]: type === ButtonType.Danger,
    [styles.default]: type === ButtonType.Default,
    [styles.primary]: type === ButtonType.Primary
  });
  return (
    <button className={buttonStyle} onClick={clickHandler} style={style}>
      {children}
    </button>
  );
};
