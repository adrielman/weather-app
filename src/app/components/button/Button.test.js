import React from "react";
import { shallow } from "enzyme";
import { Button } from "./Button";

describe("Button", () => {
  const clickHandler = jest.fn();
  const component = shallow(<Button onClick={clickHandler} />);

  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should match snapshot when compact", () => {
    const compactButton = shallow(<Button onClick={clickHandler} compact />);
    expect(compactButton).toMatchSnapshot();
  });

  it("should call handler when clicked", () => {
    component.find("button").simulate("click", { preventDefault: jest.fn() });
    expect(clickHandler).toBeCalledTimes(1);
  });
});
