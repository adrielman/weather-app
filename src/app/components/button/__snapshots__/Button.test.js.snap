// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
/>
`;

exports[`Button should match snapshot when compact 1`] = `
<button
  className="button compact"
  onClick={[Function]}
/>
`;
